client/my-sites/stats/stats-notices/all-notice-definitions.ts

```diff
@@ -34,7 +34,34 @@ const ALL_STATS_NOTICES: StatsNoticeType[] = [
 	{
 		component: CommercialSiteUpgradeNotice,
 		noticeId: 'commercial_site_upgrade',
-		isVisibleFunc: shouldShowCommercialSiteUpgradeNotice,
+		isVisibleFunc: ( {
+			isOdysseyStats,
+			isWpcom,
+			isVip,
+			isP2,
+			isOwnedByTeam51,
+			hasPaidStats,
+			isSiteJetpackNotAtomic,
+			isCommercial,
+			hasPWYWPlanOnly,
+		}: StatsNoticeProps ) => {
+			const showUpgradeNoticeForWpcomSites = isWpcom && ! isP2 && ! isOwnedByTeam51;
+			const showUpgradeNoticeForJetpackSites = isOdysseyStats || isSiteJetpackNotAtomic;
```

Comment:
The logic here is consistent with what it was; however as more simple sites are on Odyssey Stats, we probably want to change the condition to `isSiteJetpackNotAtomic && isOdysseyStats`. Some other places need tweaks too. I don't think that'll fall in the scope of the PR tho.